gwcs/api.py

```diff
@@ -185,23 +183,29 @@ def pixel_to_world(self, *pixel_arrays):
         """
         Convert pixel values to world coordinates.
         """
-        kwargs = {}
-        if not self.forward_transform.uses_quantity:
-            kwargs = {'with_units': True}
-        return self(*pixel_arrays, **kwargs)
+        if self.forward_transform.uses_quantity:
+            pixel_arrays = pixel_arrays * u.pix  # *= only for newer astropy
+        return self(*pixel_arrays, with_units=True)
 
     def array_index_to_world(self, *index_arrays):
         """
         Convert array indices to world coordinates (represented by Astropy
         objects).
         """
-        return self(*(index_arrays[::-1]), with_units=True)
+        pixel_arrays = index_arrays[::-1]
+        if self.forward_transform.uses_quantity:
+            pixel_arrays = pixel_arrays * u.pix  # *= only for newer astropy
+        return self(*pixel_arrays, with_units=True)
 
     def world_to_pixel(self, *world_objects):
         """
         Convert world coordinates to pixel values.
         """
-        return self.invert(*world_objects)
+        if not self.forward_transform.uses_quantity:
+            return self.invert(*world_objects)
+        else:
+            xpix, ypix = self.invert(*world_objects)
+            return xpix.value, ypix.value
```

Comment:
This assumes two outputs. Could you rewrite it as 
```
result = self.invert(*world_objects)
if not self.forward_transform.uses_quantity:
    result = [i.value for i in result]
```

or something similar.

Reply:
Done, thanks!